dvc/repo/get.py

```diff
@@ -66,31 +64,40 @@ def get(url, path, out=None, rev=None):
             # the same cache file might be used a few times in a directory.
             repo.cache.local.cache_types = ["reflink", "hardlink", "copy"]
 
-            output = None
-            output_error = None
-
             try:
                 output = repo.find_out_by_relpath(path)
-            except OutputNotFoundError as ex:
-                output_error = ex
-
-            is_git_file = output_error and not os.path.isabs(path)
-            is_not_cached = output and not output.use_cache
+            except OutputNotFoundError:
+                output = None
 
-            if is_git_file or is_not_cached:
-                _copy_git_file(repo, path, out, url)
-                return
+            if output and output.use_cache:
+                _get_cached(repo, output, out)
+            else:
+                # Either an uncached out with absolute path or a user error
+                if os.path.isabs(path):
+                    raise FileNotFoundError
 
-            if output_error:
-                raise OutputNotFoundError(path)
-
-            with repo.state:
-                repo.cloud.pull(output.get_used_cache())
-            output.path_info = PathInfo(os.path.abspath(out))
-            with output.repo.state:
-                output.checkout()
+                _copy(os.path.join(repo.root_dir, path), out)
 
+    except (OutputNotFoundError, FileNotFoundError):
+        raise PathMissingError(path, url)
     except NotDvcRepoError:
         raise UrlNotDvcRepoError(url)
     finally:
         remove(tmp_dir)
+
+
+def _get_cached(repo, output, out):
+    with repo.state:
+        repo.cloud.pull(output.get_used_cache())
+        output.path_info = PathInfo(os.path.abspath(out))
+        failed = output.checkout()
+        # This might happen when pull haven't really pulled all the files
+        if failed:
+            raise FileNotFoundError
```

Comment:
But which file not found, need to provide the message at least

Reply:
This would be wrapped info PathMissingError

Reply:
The message in `PathMissingError` is not really right for this case. We can't check out because cache is missing. There would be cache missing warnings, so a user may still understand what is happening.